### PR TITLE
ad7193 driver rework

### PR DESCRIPTION
**Pull request by APorumb**
AD7193 driver rework including:

-removed CS as gpio (the new SPI driver toggles automatically CS pin)
-range enum added
-op mode enum added
-clock enum added
-restyled

### CODE REVIEW

**Review by amiclaus — APPROVED** [review]

**Comment by APorumb** [comment]
Reset rework added.

-reset moved before ID check -> this way we eliminate the possibility of
polling a desynchronized spi frame device

-reset now is using memset

-modified number of bits needed for reset (Datasheet mentions 40 bits)

Signed-off-by: Andrei Porumb <andrei.porumb@analog.com>

File: drivers/adc/ad7193/ad7193.c [hunk 1]
```diff
@@ -270,15 +271,11 @@ int ad7193_set_masked_register_value(struct ad7193_dev *dev,
...
-	register_word[2] = 0xFF;
-	register_word[3] = 0xFF;
-	register_word[4] = 0xFF;
-	register_word[5] = 0xFF;
-	return spi_write_and_read(dev->spi_desc, register_word, 6);
+	uint8_t register_word[5];
+
+	memset(register_word, 0xFF, 5);
+
+	return spi_write_and_read(dev->spi_desc, register_word, 5);
```
[the remark this answers is not in the document]
Comment (APorumb) [reply drivers/adc/ad7193/ad7193.c]: Delay added, thanks for tip!

**Comment by APorumb** [comment]
Edit 1 :

-added 500 us delay after reset

**Review by buha — APPROVED** [review]
looks good, but let's only merge it when we are sure digital interface is working well

**Pull request merged**